File: src/main/java/com/pengrad/telegrambot/model/request/ReplyKeyboardMarkup.java
package com.pengrad.telegrambot.model.request;

/**
 * stas
 * 8/4/15.
 */
public class ReplyKeyboardMarkup extends Keyboard {

    private final KeyboardButton[][] keyboard;
    private boolean resize_keyboard;
    private boolean one_time_keyboard;
    private boolean selective;

    public ReplyKeyboardMarkup(String[]... keyboard) {
        this(keyboard, false, false, false);
    }

    public ReplyKeyboardMarkup(String[][] keyboard, boolean resize_keyboard, boolean one_time_keyboard, boolean selective) {
        KeyboardButton[][] keyboardButtons = new KeyboardButton[keyboard.length][];
        for (int i = 0; i < keyboard.length; i++) {
            keyboardButtons[i] = new KeyboardButton[keyboard[i].length];
            for (int j = 0; j < keyboard[i].length; j++) {
                keyboardButtons[i][j] = new KeyboardButton(keyboard[i][j]);
            }
        }
        this.keyboard = keyboardButtons;
        this.resize_keyboard = resize_keyboard;
        this.one_time_keyboard = one_time_keyboard;
        this.selective = selective;
    }

    public ReplyKeyboardMarkup(KeyboardButton[]... keyboard) {
        this.keyboard = keyboard;
    }

    public ReplyKeyboardMarkup resizeKeyboard(boolean resizeKeyboard) {
        this.resize_keyboard = resizeKeyboard;
        return this;
    }

    public ReplyKeyboardMarkup oneTimeKeyboard(boolean oneTimeKeyboard) {
        this.one_time_keyboard = oneTimeKeyboard;
        return this;
    }

    public ReplyKeyboardMarkup selective(boolean selective) {
        this.selective = selective;
        return this;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/ReplyKeyboardMarkup.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * stas
 * 8/4/15.
 */
class ReplyKeyboardMarkup(
        private val keyboard: Array<out Array<KeyboardButton>>,
        private var resize_keyboard: Boolean = false,
        private var one_time_keyboard: Boolean = false,
        private var selective: Boolean = false
) : Keyboard(), Serializable {

    constructor(vararg keyboard: Array<String>) : this(
            keyboard.map { it.map { text -> KeyboardButton(text) }.toTypedArray() }.toTypedArray()
    )

    constructor(vararg keyboard: Array<KeyboardButton>) : this(keyboard)

    fun resizeKeyboard(resizeKeyboard: Boolean): ReplyKeyboardMarkup {
        val r = ReplyKeyboardMarkup(arrayOf(KeyboardButton(""), KeyboardButton("")))
        this.resize_keyboard = resizeKeyboard
        return this
    }

    fun oneTimeKeyboard(oneTimeKeyboard: Boolean): ReplyKeyboardMarkup {
        this.one_time_keyboard = oneTimeKeyboard
        return this
    }

    fun selective(selective: Boolean): ReplyKeyboardMarkup {
        this.selective = selective
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
